src/store/asana/index.ts

```diff
@@ -550,12 +551,19 @@ async function loadTasks(action: (tasks: Task[]) => any, lastUpdated: string | n
     }
     let taskResponse: any = await asanaClient.tasks.findAll(options);
     
-    const tasks: Task[] = [];
+    let tasks: Task[] = [];
     for (; taskResponse; taskResponse = await taskResponse.nextPage()) {
       tasks.push(...taskResponse.data);
     }
-    action(tasks);
 
+    // add attachments to task
+    tasks = await Promise.all(tasks.map(async (task) => {
```

Comment:
What sort of performance difference are we seeing with this? Trying to get the task attachments for all tasks feels like it could be slow

Reply:
It's noticeably slower for sure. I'm just not sure what alternatives we have other than getting the attachments for all tasks (since we need to display these images for each task). Do you have other ideas?

Reply:
I think first step would be removing the await on the Promise.all; that is going to essentially block all other actions. Having it run in the background is probably fine and will have less noticeable performance impact